modules/internal/String.chpl

```diff
@@ -1587,6 +1587,32 @@ module String {
       return _join(S);
     }
 
+    /*
+      :args listToJoin: list contaning elements whose string values will be concatenated
+      .. code-block:: chapel
+      var x = "|".join([1,2,3])
+      writeln(x); //prints "a|10|d"
+    
+    */
+
+    proc join(listToJoin: list) : string {
```

Comment:
We'll want to further instantiate the type of the arg `listToJoin` so that we restrict it to a list of `string`. That would look something like:

`proc join(listToJoin: list(string)): string { ... }`